clang-tools-extra/docs/clang-tidy/checks/bugprone/virtual-arithmetic.rst

```diff
@@ -0,0 +1,50 @@
+.. title:: clang-tidy - bugprone-virtual-arithmetic
+
+bugprone-virtual-arithmetic
+===========================
+
+Warn if pointer arithmetic is performed on a class that declares a
+virtual function.
+
+Pointer arithmetic on polymorphic objects where the pointer's static type is 
+different from its dynamic type is undefined behavior.
```

Comment:
this apply not only to polymorphic objects. Would be good to write why (maybe something about pointer to virtual table).

Reply:
make sure just to write that it may cause crash as due to size mismatch vtable pointer may point to invalid  address.

Reply:
Added.